File: Assets/Plugins/Android/src/com/playphone/multinet/unity/MNDirectUIHelperUnity.java
package com.playphone.multinet.unity;

import com.playphone.multinet.*;
import com.unity3d.player.UnityPlayer;

public class MNDirectUIHelperUnity {
    public static void setDashboardStyle(final int newStyle) {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirectUIHelper.setDashboardStyle(newStyle);
            }
        });
    }

    public static int getDashboardStyle() {
        MNUnity.MARK();

        return MNDirectUIHelper.getDashboardStyle();
    }

    public static void showDashboard() {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirectUIHelper.showDashboard();
            }
        });
    }

    public static void hideDashboard() {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirectUIHelper.hideDashboard();
            }
        });
    }

    public static boolean isDashboardHidden() {
        MNUnity.MARK();

        return MNDirectUIHelper.isDashboardHidden();
    }

    public static boolean isDashboardVisible() {
        MNUnity.MARK();

        return MNDirectUIHelper.isDashboardVisible();
    }

    public static void setHostActivity(final boolean setFlag) {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                if (setFlag) {
                    MNDirectUIHelper.setHostActivity(UnityPlayer.currentActivity);
                }
                else {
                    MNDirectUIHelper.setHostActivity(null);
                }
            }
        });
    }

    protected static class MNDirectUIHelperEventHandler extends com.playphone.multinet.MNDirectUIHelper.EventHandlerAbstract {

        @Override
        public void onShowDashboard() {
            MNUnity.MARK();

            UnityPlayer.currentActivity.runOnUiThread(new Runnable()
            {
                public void run()
                {
                  MNUnity.callUnityFunction("MNUM_onShowDashboard");
                }
            });
        }

        @Override
        public void onHideDashboard() {
            MNUnity.MARK();

            UnityPlayer.currentActivity.runOnUiThread(new Runnable()
            {
                public void run()
                {
                  MNUnity.callUnityFunction("MNUM_onHideDashboard");
                }
            });
        }
    }

    private static MNDirectUIHelperEventHandler registeredEventHandler = null;
    public static synchronized boolean registerEventHandler() {
        MNUnity.MARK();

        if (registeredEventHandler == null) {
            registeredEventHandler = new MNDirectUIHelperUnity.MNDirectUIHelperEventHandler();
            MNDirectUIHelper.addEventHandler(registeredEventHandler);
        }

        return true;
    }
    
    public static synchronized boolean unregisterEventHandler() {
        MNUnity.MARK();

        if (registeredEventHandler != null) {
            MNDirectUIHelper.removeEventHandler(registeredEventHandler);
            registeredEventHandler = null;
        }

        return true;
    }
}

File: Assets/Plugins/iOS/PlayPhoneMultiNet/unity/MNDirectUIHelperUnity.m
//
//  MNDirectUIHelperUnity.m
//  PlayPhone SDK for Unity
//
//  Copyright (c) 2012 PlayPhone Inc. All rights reserved.
//

#import "MNDirectUIHelper.h"

#import "MNDirectUIHelperUnity.h"

@interface MNDirectUIHelperUnity : NSObject <MNDirectUIHelperDelegate>
+ (id)shared;
- (void)mnUIHelperDashboardShown;
- (void)mnUIHelperDashboardHidden;
@end

static MNDirectUIHelperUnity *MNDirectUIHelperUnityInstance = nil;

@implementation MNDirectUIHelperUnity

+(id)shared {
    if (MNDirectUIHelperUnityInstance == nil) {
        MNDirectUIHelperUnityInstance = [[MNDirectUIHelperUnity alloc]init];
    }
    
    return MNDirectUIHelperUnityInstance;
}

- (void)mnUIHelperDashboardShown {
    [MNUnity callUnityFunction:@"MNUM_onShowDashboard" withParams:nil];
}
- (void)mnUIHelperDashboardHidden {
    [MNUnity callUnityFunction:@"MNUM_onHideDashboard" withParams:nil];
}
@end

EXTERN_C
void _MNDirectUIHelper_SetDashboardStyle(int newStyle) {
    MARK;

    [MNDirectUIHelper setDashboardStyle:newStyle];
}

EXTERN_C
int _MNDirectUIHelper_GetDashboardStyle() {
    MARK;

    return [MNDirectUIHelper getDashboardStyle];
}

EXTERN_C
void _MNDirectUIHelper_ShowDashboard() {
    MARK;

    [MNDirectUIHelper showDashboard];
}

EXTERN_C
void _MNDirectUIHelper_HideDashboard() {
    MARK;

    [MNDirectUIHelper hideDashboard];
}

EXTERN_C
bool _MNDirectUIHelper_IsDashboardHidden() {
    MARK;

    return [MNDirectUIHelper isDashboardHidden];
}

EXTERN_C
bool _MNDirectUIHelper_IsDashboardVisible() {
    MARK;

    return [MNDirectUIHelper isDashboardVisible];
}


EXTERN_C
bool _MNDirectUIHelper_RegisterEventHandler(){
    MARK;

    [MNDirectUIHelper addDelegate:[MNDirectUIHelperUnity shared]];

    return true;
}

EXTERN_C
bool _MNDirectUIHelper_UnregisterEventHandler(){
    MARK;

    [MNDirectUIHelper removeDelegate:[MNDirectUIHelperUnity shared]];

    return true;
}


